starlette/middleware/cors.py

```diff
@@ -131,9 +132,9 @@ def preflight_response(self, request_headers: Headers) -> Response:
         # if we do.
         if failures:
             failure_text = "Disallowed CORS " + ", ".join(failures)
-            return PlainTextResponse(failure_text, status_code=400, headers=headers)
+            return PlainTextResponse(failure_text, status_code=400, headers=dict(headers))
 
-        return PlainTextResponse("OK", status_code=200, headers=headers)
+        return PlainTextResponse("OK", status_code=200, headers=dict(headers))
```

Comment:
Ugh, I don't like this. `Response` marks `headers` as type `dict`. I could go and update all of the `Response` classes to take `Mapping[Any, Any]` or `Mapping[str, str]`, or even some `Union` type with `Mapping` and `dict`, but that seems way out of scope for this PR.

Maybe I should just revert this and not use `allow_explicit_origin()` in the preflight case, since that's what requires a `MutableHeaders` object?